src/Bicep.Core/Features/FeatureProvider.cs

```diff
@@ -22,9 +23,17 @@ public string CacheRootDirectory
             }
         }
 
-        public bool RegistryEnabled => bool.TryParse(Environment.GetEnvironmentVariable("BICEP_REGISTRY_ENABLED_EXPERIMENTAL"), out var enabled) ? enabled : false;
+        public bool RegistryEnabled => ReadBooleanEnvVar("BICEP_REGISTRY_ENABLED_EXPERIMENTAL", defaultValue: false);
 
-        public bool SymbolicNameCodegenEnabled => bool.TryParse(Environment.GetEnvironmentVariable("BICEP_SYMBOLIC_NAME_CODEGEN_EXPERIMENTAL"), out var enabled) ? enabled : false;
+        public bool SymbolicNameCodegenEnabled => ReadBooleanEnvVar("BICEP_SYMBOLIC_NAME_CODEGEN_EXPERIMENTAL", defaultValue: false);
+
+        private Lazy<bool> importsEnabledLazy = new(() => ReadBooleanEnvVar("BICEP_IMPORTS_ENABLED_EXPERIMENTAL", defaultValue: false), LazyThreadSafetyMode.PublicationOnly);
+        public bool ImportsEnabled => importsEnabledLazy.Value;
```

Comment:
>importsEnabledLazy [](http://example.com/codeflow?start=38&length=18)

Do we need to switch the others to use `Lazy<T>`?

Reply:
Sure - probably worth it for consistency. This one is potentially read multiple times during a compilation, which is why I used `Lazy`. The other option would be to just set them all in the constructor.